src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -429,12 +441,11 @@ applyUpgrades v cid height
 
       infoLog $ "Running upgrade tx " <> sshow (_cmdHash tx)
 
-      r <- try $ runGenesis tx permissiveNamespacePolicy interp
-
-      case r of
+      tryAllSynchronous (runGenesis tx permissiveNamespacePolicy interp) >>= \case
         Right _ -> return ()
-        Left (e :: SomeException) -> do
+        Left e -> do
           logError $ "Upgrade transaction failed! " <> sshow e
+            -- TODO: is it really OK, to just swallow the exception?
```

Comment:
@emilypi What is your though on this comment?

Reply:
It would be pretty catastrophic if the upgrades didn't work, but other parts of the service would immediately begin to fail block validation as a result, so we end up with an error being thrown either way. We may as well just throw an internal error here.

Reply:
I'd say, if it is catastrophic, let's throw here so here and don't wait for other parts falling apart. That should help diagnosing the issue, if it ever happens.